compiler/noirc_frontend/src/hir/def_collector/dc_mod.rs

```diff
@@ -322,10 +333,18 @@ impl<'a> ModCollector<'a> {
             // And store the TypeId -> StructType mapping somewhere it is reachable
             self.def_collector.items.types.insert(id, unresolved);
 
-            context.def_interner.add_definition_location(
-                ReferenceId::Struct(id),
-                Some(ModuleId { krate, local_id: self.module_id }),
-            );
+            if context.def_interner.is_in_lsp_mode() {
+                context.def_interner.add_definition_location(
+                    ReferenceId::Struct(id),
+                    Some(ModuleId { krate, local_id: self.module_id }),
+                );
+
+                context.def_interner.register_name_for_auto_import(
+                    name.to_string(),
+                    ModuleDefId::TypeId(id),
+                    ItemVisibility::Public,
+                );
```

Comment:
Does add_definition_location imply register_name_for_auto_import? If so maybe there could be a helper.

Reply:
I initially thought about doing that. But... we also do `add_definition_location` for impl methods and trait impl methods, struct members, and I think local variables too. So it's not always that they need to be done together.

That said, maybe they could be done together but the name for auto import could be an `Option` that's passed to `add_definition_location`?

Reply:
Sounds good to me, or optionally another wrapper that calls both functions that you only need to call in a couple cases.

Reply:
I ended up introducing several wrappers, mainly because in one place a `ModuleDefId` is used and in another place a `ReferenceId` is used. That said, I think it looks much better than now (less code, and more readable).